src/Form/GraphQLQueryMapForm.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Drupal\graphql\Form;
+
+use Drupal\Core\Entity\EntityForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Form controller for GraphQL query map forms.
+ */
+class GraphQLQueryMapForm extends EntityForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+
+    $query_map = $this->entity;
+
+    $form['#title'] = $this->t('Edit %title GraphQL Query Map', ['%title' => $query_map->id()]);
+
+    $form['version_id'] = [
+      '#type' => 'machine_name',
+      '#default_value' => $query_map->id(),
+      '#title' => $this->t('Version ID'),
+      '#machine_name' => [
+        'exists' => ['\Drupal\graphql\Entity\GraphQLQueryMap', 'exists'],
+      ],
+      '#description' => 'The SHA1 hash of the queries.json file.',
+      '#disabled' => !$query_map->isNew(),
+    ];
+
+    $form['query_map'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('GraphQL Queries'),
+      '#prefix' => '<div id="names-fieldset-wrapper">',
+      '#suffix' => '</div>',
+      '#tree' => TRUE,
+    ];
+
+    $count = $form_state->get('query_map_count');
+    if (empty($count)) {
+      if ($query_map->isNew()) {
+        $count = 1;
+        $form_state->set('query_map_count', 1);
+      }
+      else {
+        $count = count($query_map->query_map);
+        $form_state->set('query_map_count', $count);
+      }
+    }
+
+    for ($i = 1; $i <= $count; $i++) {
+      $form['query_map'][$i] = [
+        '#type' => 'textarea',
+        '#title' => $this->t('Query'),
+        '#rows' => 10,
+        '#description' => $this->t('Enter one graphql query.'),
+        '#default_value' => !$query_map->isNew() ? ($query_map->query_map[$i] ?: '') : '',
+      ];
+    }
+    $form['query_map']['actions']['add_name'] = [
+      '#type' => 'submit',
+      '#value' => t('Add one more'),
+      '#submit' => array('::addOne'),
+      '#ajax' => [
+        'callback' => '::addmoreCallback',
+        'wrapper' => 'names-fieldset-wrapper',
+      ],
+    ];
+    if ($count > 1) {
+      $form['query_map']['actions']['remove_name'] = [
+        '#type' => 'submit',
+        '#value' => t('Remove one'),
+        '#submit' => array('::removeCallback'),
+        '#ajax' => [
+          'callback' => '::addmoreCallback',
+          'wrapper' => 'names-fieldset-wrapper',
+        ]
+      ];
+    }
+
+    $form['actions'] = [
+      '#type' => 'actions',
+    ];
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Submit'),
+    ];
+
+    return $form;
+  }
+
+  public function addOne(array &$form, FormStateInterface $form_state) {
```

Comment:
This is nice but realistically we would have an upload of a .json file and that would then be parsed and the queries from that json file woudl be added one by one.